{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5",
    "execution": {
     "iopub.execute_input": "2025-11-09T09:43:49.264596Z",
     "iopub.status.busy": "2025-11-09T09:43:49.264226Z",
     "iopub.status.idle": "2025-11-09T09:44:07.224964Z",
     "shell.execute_reply": "2025-11-09T09:44:07.223472Z",
     "shell.execute_reply.started": "2025-11-09T09:43:49.264559Z"
    },
    "trusted": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using Kaggle's public dataset BigQuery integration.\n"
     ]
    }
   ],
   "source": [
    "from google.cloud import bigquery\n",
    "\n",
    "client = bigquery.Client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.227677Z",
     "iopub.status.busy": "2025-11-09T09:44:07.226771Z",
     "iopub.status.idle": "2025-11-09T09:44:07.295484Z",
     "shell.execute_reply": "2025-11-09T09:44:07.294187Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.227621Z"
    },
    "trusted": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo Symbol</th>\n",
       "      <th>Entrez Gene ID</th>\n",
       "      <th>GRCh37 Isoform</th>\n",
       "      <th>GRCh37 RefSeq</th>\n",
       "      <th>GRCh38 Isoform</th>\n",
       "      <th>GRCh38 RefSeq</th>\n",
       "      <th>Is Oncogene</th>\n",
       "      <th>Is Tumor Suppressor Gene</th>\n",
       "      <th># of occurrence within resources (Column J-P)</th>\n",
       "      <th>OncoKB Annotated</th>\n",
       "      <th>MSK-IMPACT</th>\n",
       "      <th>MSK-HEME</th>\n",
       "      <th>FOUNDATION ONE</th>\n",
       "      <th>FOUNDATION ONE HEME</th>\n",
       "      <th>Vogelstein</th>\n",
       "      <th>COSMIC CGC (v99)</th>\n",
       "      <th>Gene Aliases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>ENST00000318560</td>\n",
       "      <td>NM_005157.4</td>\n",
       "      <td>ENST00000318560</td>\n",
       "      <td>NM_005157.4</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>7</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ABL, JTK7, c-ABL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>ENST00000349310</td>\n",
       "      <td>NM_001014431.1</td>\n",
       "      <td>ENST00000349310</td>\n",
       "      <td>NM_001014431.1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>7</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>AKT, PKB, PRKBA, RAC, RAC-alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>ENST00000389048</td>\n",
       "      <td>NM_004304.4</td>\n",
       "      <td>ENST00000389048</td>\n",
       "      <td>NM_004304.4</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>7</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CD246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMER1</td>\n",
       "      <td>139285</td>\n",
       "      <td>ENST00000330258</td>\n",
       "      <td>NM_152424.3</td>\n",
       "      <td>ENST00000374869</td>\n",
       "      <td>NM_152424.3</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>7</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>FAM123B, FLJ39827, RP11-403E24.2, WTX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>APC</td>\n",
       "      <td>324</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>NM_000038.5</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>NM_000038.5</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>7</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>DP2.5, PPP1R46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo Symbol  Entrez Gene ID   GRCh37 Isoform   GRCh37 RefSeq  \\\n",
       "0        ABL1              25  ENST00000318560     NM_005157.4   \n",
       "1        AKT1             207  ENST00000349310  NM_001014431.1   \n",
       "2         ALK             238  ENST00000389048     NM_004304.4   \n",
       "3       AMER1          139285  ENST00000330258     NM_152424.3   \n",
       "4         APC             324  ENST00000257430     NM_000038.5   \n",
       "\n",
       "    GRCh38 Isoform   GRCh38 RefSeq Is Oncogene Is Tumor Suppressor Gene  \\\n",
       "0  ENST00000318560     NM_005157.4         Yes                       No   \n",
       "1  ENST00000349310  NM_001014431.1         Yes                       No   \n",
       "2  ENST00000389048     NM_004304.4         Yes                       No   \n",
       "3  ENST00000374869     NM_152424.3          No                      Yes   \n",
       "4  ENST00000257430     NM_000038.5          No                      Yes   \n",
       "\n",
       "   # of occurrence within resources (Column J-P) OncoKB Annotated MSK-IMPACT  \\\n",
       "0                                              7              Yes        Yes   \n",
       "1                                              7              Yes        Yes   \n",
       "2                                              7              Yes        Yes   \n",
       "3                                              7              Yes        Yes   \n",
       "4                                              7              Yes        Yes   \n",
       "\n",
       "  MSK-HEME FOUNDATION ONE FOUNDATION ONE HEME Vogelstein COSMIC CGC (v99)  \\\n",
       "0      Yes            Yes                 Yes        Yes              Yes   \n",
       "1      Yes            Yes                 Yes        Yes              Yes   \n",
       "2      Yes            Yes                 Yes        Yes              Yes   \n",
       "3      Yes            Yes                 Yes        Yes              Yes   \n",
       "4      Yes            Yes                 Yes        Yes              Yes   \n",
       "\n",
       "                            Gene Aliases  \n",
       "0                       ABL, JTK7, c-ABL  \n",
       "1        AKT, PKB, PRKBA, RAC, RAC-alpha  \n",
       "2                                  CD246  \n",
       "3  FAM123B, FLJ39827, RP11-403E24.2, WTX  \n",
       "4                         DP2.5, PPP1R46  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "\n",
    "gene_list = pd.read_csv('/kaggle/input/oncokb-cancer-gene-list/cancerGeneList.tsv', sep='\\t')\n",
    "gene_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.367611Z",
     "iopub.status.busy": "2025-11-09T09:44:07.367101Z",
     "iopub.status.idle": "2025-11-09T09:44:07.644451Z",
     "shell.execute_reply": "2025-11-09T09:44:07.643096Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.367552Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "df = pd.read_csv('https://raw.githubusercontent.com/marcin119a/data/refs/heads/main/bq-metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.646291Z",
     "iopub.status.busy": "2025-11-09T09:44:07.645885Z",
     "iopub.status.idle": "2025-11-09T09:44:07.672533Z",
     "shell.execute_reply": "2025-11-09T09:44:07.671046Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.646254Z"
    },
    "trusted": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater\n",
      "  has_large_values = (abs_vals > 1e6).any()\n",
      "/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less\n",
      "  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()\n",
      "/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater\n",
      "  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>BQ Project</th>\n",
       "      <th>BQ Dataset</th>\n",
       "      <th>BQ Table</th>\n",
       "      <th>Full ID</th>\n",
       "      <th>Program</th>\n",
       "      <th>Category</th>\n",
       "      <th>Reference Genome</th>\n",
       "      <th>Source</th>\n",
       "      <th>Data Type</th>\n",
       "      <th>Exp. Strategy</th>\n",
       "      <th>Status</th>\n",
       "      <th>Rows</th>\n",
       "      <th>Created</th>\n",
       "      <th>Example Joins</th>\n",
       "      <th>Description</th>\n",
       "      <th>Labels</th>\n",
       "      <th>Fields</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...</td>\n",
       "      <td>isb-cgc-bq</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>quant_phosphoproteome_academia_sinica_LUAD100_...</td>\n",
       "      <td>`isb-cgc-bq.ICPC.quant_phosphoproteome_academi...</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>PROCESSED -OMICS DATA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDC</td>\n",
       "      <td>PHOSPHOPROTEOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CURRENT</td>\n",
       "      <td>11,667,348</td>\n",
       "      <td>9/23/2022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normalized log2ratio protein abundance estimat...</td>\n",
       "      <td>access:open, category:processed_-omics_data, d...</td>\n",
       "      <td>case_id, sample_id, aliquot_id, aliquot_submit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...</td>\n",
       "      <td>isb-cgc-bq</td>\n",
       "      <td>ICPC_versioned</td>\n",
       "      <td>quant_phosphoproteome_academia_sinica_LUAD100_...</td>\n",
       "      <td>`isb-cgc-bq.ICPC_versioned.quant_phosphoproteo...</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>PROCESSED -OMICS DATA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDC</td>\n",
       "      <td>PHOSPHOPROTEOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CURRENT</td>\n",
       "      <td>11,667,348</td>\n",
       "      <td>9/23/2022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normalized log2ratio protein abundance estimat...</td>\n",
       "      <td>access:open, category:processed_-omics_data, d...</td>\n",
       "      <td>case_id, sample_id, aliquot_id, aliquot_submit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...</td>\n",
       "      <td>isb-cgc-bq</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>quant_proteome_academia_sinica_LUAD100_pdc_cur...</td>\n",
       "      <td>`isb-cgc-bq.ICPC.quant_proteome_academia_sinic...</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>PROCESSED -OMICS DATA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDC</td>\n",
       "      <td>PROTEIN EXPRESSION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CURRENT</td>\n",
       "      <td>5,447,088</td>\n",
       "      <td>2/27/2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normalized log2ratio protein abundance estimat...</td>\n",
       "      <td>access:open, category:processed_-omics_data, d...</td>\n",
       "      <td>case_id, sample_id, aliquot_id, aliquot_submit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...</td>\n",
       "      <td>isb-cgc-bq</td>\n",
       "      <td>ICPC_versioned</td>\n",
       "      <td>quant_proteome_academia_sinica_LUAD100_pdc_V2_16</td>\n",
       "      <td>`isb-cgc-bq.ICPC_versioned.quant_proteome_acad...</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>PROCESSED -OMICS DATA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDC</td>\n",
       "      <td>PROTEIN EXPRESSION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CURRENT</td>\n",
       "      <td>5,447,088</td>\n",
       "      <td>2/27/2023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normalized log2ratio protein abundance estimat...</td>\n",
       "      <td>access:open, category:processed_-omics_data, d...</td>\n",
       "      <td>case_id, sample_id, aliquot_id, aliquot_submit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACADEMIA SINICA LUAD-100 CLINICAL DATA</td>\n",
       "      <td>isb-cgc-bq</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>clinical_academia_sinica_LUAD_100_pdc_current</td>\n",
       "      <td>`isb-cgc-bq.ICPC.clinical_academia_sinica_LUAD...</td>\n",
       "      <td>ICPC</td>\n",
       "      <td>CLINICAL BIOSPECIMEN DATA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDC</td>\n",
       "      <td>CLINICAL DATA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CURRENT</td>\n",
       "      <td>110</td>\n",
       "      <td>5/27/2021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data for the Academia Sinica LUAD-100 project ...</td>\n",
       "      <td>access:open, category:clinical_biospecimen_dat...</td>\n",
       "      <td>case_id, case_submitter_id, project_submitter_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Name  BQ Project  \\\n",
       "0  ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...  isb-cgc-bq   \n",
       "1  ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...  isb-cgc-bq   \n",
       "2  ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...  isb-cgc-bq   \n",
       "3  ACADEMIA SINICA LUAD-100 (LUNG ADENOCARCINOMA)...  isb-cgc-bq   \n",
       "4             ACADEMIA SINICA LUAD-100 CLINICAL DATA  isb-cgc-bq   \n",
       "\n",
       "       BQ Dataset                                           BQ Table  \\\n",
       "0            ICPC  quant_phosphoproteome_academia_sinica_LUAD100_...   \n",
       "1  ICPC_versioned  quant_phosphoproteome_academia_sinica_LUAD100_...   \n",
       "2            ICPC  quant_proteome_academia_sinica_LUAD100_pdc_cur...   \n",
       "3  ICPC_versioned   quant_proteome_academia_sinica_LUAD100_pdc_V2_16   \n",
       "4            ICPC      clinical_academia_sinica_LUAD_100_pdc_current   \n",
       "\n",
       "                                             Full ID Program  \\\n",
       "0  `isb-cgc-bq.ICPC.quant_phosphoproteome_academi...    ICPC   \n",
       "1  `isb-cgc-bq.ICPC_versioned.quant_phosphoproteo...    ICPC   \n",
       "2  `isb-cgc-bq.ICPC.quant_proteome_academia_sinic...    ICPC   \n",
       "3  `isb-cgc-bq.ICPC_versioned.quant_proteome_acad...    ICPC   \n",
       "4  `isb-cgc-bq.ICPC.clinical_academia_sinica_LUAD...    ICPC   \n",
       "\n",
       "                    Category Reference Genome Source           Data Type  \\\n",
       "0      PROCESSED -OMICS DATA              NaN    PDC     PHOSPHOPROTEOME   \n",
       "1      PROCESSED -OMICS DATA              NaN    PDC     PHOSPHOPROTEOME   \n",
       "2      PROCESSED -OMICS DATA              NaN    PDC  PROTEIN EXPRESSION   \n",
       "3      PROCESSED -OMICS DATA              NaN    PDC  PROTEIN EXPRESSION   \n",
       "4  CLINICAL BIOSPECIMEN DATA              NaN    PDC       CLINICAL DATA   \n",
       "\n",
       "  Exp. Strategy   Status        Rows    Created  Example Joins  \\\n",
       "0           NaN  CURRENT  11,667,348  9/23/2022            NaN   \n",
       "1           NaN  CURRENT  11,667,348  9/23/2022            NaN   \n",
       "2           NaN  CURRENT   5,447,088  2/27/2023            NaN   \n",
       "3           NaN  CURRENT   5,447,088  2/27/2023            NaN   \n",
       "4           NaN  CURRENT         110  5/27/2021            NaN   \n",
       "\n",
       "                                         Description  \\\n",
       "0  Normalized log2ratio protein abundance estimat...   \n",
       "1  Normalized log2ratio protein abundance estimat...   \n",
       "2  Normalized log2ratio protein abundance estimat...   \n",
       "3  Normalized log2ratio protein abundance estimat...   \n",
       "4  Data for the Academia Sinica LUAD-100 project ...   \n",
       "\n",
       "                                              Labels  \\\n",
       "0  access:open, category:processed_-omics_data, d...   \n",
       "1  access:open, category:processed_-omics_data, d...   \n",
       "2  access:open, category:processed_-omics_data, d...   \n",
       "3  access:open, category:processed_-omics_data, d...   \n",
       "4  access:open, category:clinical_biospecimen_dat...   \n",
       "\n",
       "                                              Fields  \n",
       "0  case_id, sample_id, aliquot_id, aliquot_submit...  \n",
       "1  case_id, sample_id, aliquot_id, aliquot_submit...  \n",
       "2  case_id, sample_id, aliquot_id, aliquot_submit...  \n",
       "3  case_id, sample_id, aliquot_id, aliquot_submit...  \n",
       "4  case_id, case_submitter_id, project_submitter_...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.674273Z",
     "iopub.status.busy": "2025-11-09T09:44:07.673827Z",
     "iopub.status.idle": "2025-11-09T09:44:07.695797Z",
     "shell.execute_reply": "2025-11-09T09:44:07.694456Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.674210Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "datasets_expression = df[(df['Full ID'].str.contains('RNAseq_hg38_gdc_current')) | (df['Full ID'].str.contains('Expression'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.699608Z",
     "iopub.status.busy": "2025-11-09T09:44:07.699177Z",
     "iopub.status.idle": "2025-11-09T09:44:07.718865Z",
     "shell.execute_reply": "2025-11-09T09:44:07.717549Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.699571Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "datasets_mutations = df[(df['Full ID'].str.contains('somatic')) & (df['Full ID'].str.contains('current'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.721060Z",
     "iopub.status.busy": "2025-11-09T09:44:07.720646Z",
     "iopub.status.idle": "2025-11-09T09:44:07.756601Z",
     "shell.execute_reply": "2025-11-09T09:44:07.755299Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.721023Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "merged_df = pd.merge(datasets_expression, datasets_mutations, on='BQ Dataset', suffixes=('_expr', '_mut'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.758060Z",
     "iopub.status.busy": "2025-11-09T09:44:07.757674Z",
     "iopub.status.idle": "2025-11-09T09:44:07.783379Z",
     "shell.execute_reply": "2025-11-09T09:44:07.781980Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.758029Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "filtered_df = merged_df[merged_df['BQ Table_expr'] != 'miRNAseq_hg38_gdc_current']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.785391Z",
     "iopub.status.busy": "2025-11-09T09:44:07.784906Z",
     "iopub.status.idle": "2025-11-09T09:44:07.810705Z",
     "shell.execute_reply": "2025-11-09T09:44:07.809209Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.785342Z"
    },
    "trusted": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'project_short_name, primary_site, case_barcode, sample_barcode, aliquot_barcode, gene_name, gene_type, Ensembl_gene_id, Ensembl_gene_id_v, unstranded, stranded_first, stranded_second, tpm_unstranded, fpkm_unstranded, fpkm_uq_unstranded, sample_type_name, case_gdc_id, sample_gdc_id, aliquot_gdc_id, file_gdc_id, platform,'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df['Fields_expr'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.812622Z",
     "iopub.status.busy": "2025-11-09T09:44:07.812245Z",
     "iopub.status.idle": "2025-11-09T09:44:07.832570Z",
     "shell.execute_reply": "2025-11-09T09:44:07.831061Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.812590Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "genes = ', '.join(f'\"{gene}\"' for gene in gene_list['Hugo Symbol'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T09:44:07.834649Z",
     "iopub.status.busy": "2025-11-09T09:44:07.834245Z",
     "iopub.status.idle": "2025-11-09T10:10:12.217780Z",
     "shell.execute_reply": "2025-11-09T10:10:12.216291Z",
     "shell.execute_reply.started": "2025-11-09T09:44:07.834616Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "# Tworzenie jednego zapytania z UNION ALL dla wszystkich datasetów\n",
    "union_queries = []\n",
    "for dataset in filtered_df['Full ID_expr']:\n",
    "    union_queries.append(f\"\"\"\n",
    "    SELECT gene_name, case_barcode, tpm_unstranded, primary_site\n",
    "    FROM {dataset}\n",
    "    WHERE gene_name in ({genes})\n",
    "    \"\"\")\n",
    "\n",
    "# Łączenie wszystkich zapytań w jedno z agregacją\n",
    "query = f\"\"\"\n",
    "SELECT \n",
    "    case_barcode,\n",
    "    ARRAY_AGG(gene_name ORDER BY gene_name) as gene_name,\n",
    "    ARRAY_AGG(tpm_unstranded ORDER BY gene_name) as tpm_unstranded,\n",
    "    ANY_VALUE(primary_site) as primary_site\n",
    "FROM (\n",
    "    {' UNION ALL '.join(union_queries)}\n",
    ")\n",
    "GROUP BY case_barcode\n",
    "ORDER BY case_barcode\n",
    "\"\"\"\n",
    "\n",
    "query_job = client.query(query)\n",
    "df_expressions = query_job.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_barcode</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>tpm_unstranded</th>\n",
       "      <th>primary_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [case_barcode, gene_name, tpm_unstranded, primary_site]\n",
       "Index: []"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_expressions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-11-09T10:10:14.221517Z",
     "iopub.status.busy": "2025-11-09T10:10:14.220918Z",
     "iopub.status.idle": "2025-11-09T10:10:32.203743Z",
     "shell.execute_reply": "2025-11-09T10:10:32.202481Z",
     "shell.execute_reply.started": "2025-11-09T10:10:14.221449Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "df_expressions.to_parquet('expression_onko_db.parquet')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-10-15T14:12:54.465754Z",
     "iopub.status.busy": "2025-10-15T14:12:54.465357Z",
     "iopub.status.idle": "2025-10-15T14:13:32.363053Z",
     "shell.execute_reply": "2025-10-15T14:13:32.361794Z",
     "shell.execute_reply.started": "2025-10-15T14:12:54.465726Z"
    },
    "trusted": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_barcode</th>\n",
       "      <th>tpm_unstranded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>01BR001</td>\n",
       "      <td>[0.2494, 64.2557, 11.4315, 7.1285, 1.9811, 8.7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01BR008</td>\n",
       "      <td>[1.7251, 38.5475, 10.3734, 6.9426, 1.1576, 3.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>01BR009</td>\n",
       "      <td>[1.9202, 57.0094, 11.9917, 8.1604, 1.4472, 36....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01BR010</td>\n",
       "      <td>[3.4115, 41.4999, 33.4159, 6.2088, 2.1342, 29....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01BR015</td>\n",
       "      <td>[3.2072, 55.4033, 33.4278, 9.3306, 3.807, 21.7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19555</th>\n",
       "      <td>TCGA-ZS-A9CF</td>\n",
       "      <td>[4.769, 15.642, 18.4512, 24.2891, 22.7954, 19....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19556</th>\n",
       "      <td>TCGA-ZS-A9CG</td>\n",
       "      <td>[79.5957, 18.0213, 16.5968, 2.894, 0.8658, 6.6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19557</th>\n",
       "      <td>TCGA-ZT-A8OM</td>\n",
       "      <td>[0.5088, 15.2741, 11.3042, 2.7836, 1.0875, 2.9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19558</th>\n",
       "      <td>TCGA-ZU-A8S4</td>\n",
       "      <td>[12.9105, 10.5616, 9.3254, 66.5383, 5.1272, 60...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19559</th>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>[3.8986, 59.159, 42.7654, 7.7885, 2.0867, 131....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19560 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       case_barcode                                     tpm_unstranded\n",
       "0           01BR001  [0.2494, 64.2557, 11.4315, 7.1285, 1.9811, 8.7...\n",
       "1           01BR008  [1.7251, 38.5475, 10.3734, 6.9426, 1.1576, 3.3...\n",
       "2           01BR009  [1.9202, 57.0094, 11.9917, 8.1604, 1.4472, 36....\n",
       "3           01BR010  [3.4115, 41.4999, 33.4159, 6.2088, 2.1342, 29....\n",
       "4           01BR015  [3.2072, 55.4033, 33.4278, 9.3306, 3.807, 21.7...\n",
       "...             ...                                                ...\n",
       "19555  TCGA-ZS-A9CF  [4.769, 15.642, 18.4512, 24.2891, 22.7954, 19....\n",
       "19556  TCGA-ZS-A9CG  [79.5957, 18.0213, 16.5968, 2.894, 0.8658, 6.6...\n",
       "19557  TCGA-ZT-A8OM  [0.5088, 15.2741, 11.3042, 2.7836, 1.0875, 2.9...\n",
       "19558  TCGA-ZU-A8S4  [12.9105, 10.5616, 9.3254, 66.5383, 5.1272, 60...\n",
       "19559  TCGA-ZX-AA5X  [3.8986, 59.159, 42.7654, 7.7885, 2.0867, 131....\n",
       "\n",
       "[19560 rows x 2 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_expressions_sorted = df_expressions.sort_values(by='gene_name')\n",
    "    \n",
    "# Group by case_barcode and aggregate tpm_unstranded into a list\n",
    "grouped_expressions_df = df_expressions_sorted.groupby('case_barcode')['tpm_unstranded'].apply(list).reset_index()\n",
    "\n",
    "grouped_expressions_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-10-15T14:13:54.172672Z",
     "iopub.status.busy": "2025-10-15T14:13:54.172258Z",
     "iopub.status.idle": "2025-10-15T14:13:54.186640Z",
     "shell.execute_reply": "2025-10-15T14:13:54.185621Z",
     "shell.execute_reply.started": "2025-10-15T14:13:54.172641Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "filtered_grouped_expressions_df = grouped_expressions_df[\n",
    "        grouped_expressions_df['tpm_unstranded'].apply(len) == 1180\n",
    "    ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-10-15T14:14:43.634855Z",
     "iopub.status.busy": "2025-10-15T14:14:43.634385Z",
     "iopub.status.idle": "2025-10-15T14:14:43.682023Z",
     "shell.execute_reply": "2025-10-15T14:14:43.680311Z",
     "shell.execute_reply.started": "2025-10-15T14:14:43.634814Z"
    },
    "trusted": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_barcode</th>\n",
       "      <th>tpm_unstranded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9049</th>\n",
       "      <td>TCGA-02-0003</td>\n",
       "      <td>[4.5317, 23.5094, 36.0164, 8.1348, 1.204, 38.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9050</th>\n",
       "      <td>TCGA-02-0016</td>\n",
       "      <td>[24.0906, 44.8204, 23.6335, 12.3301, 3.2369, 6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9051</th>\n",
       "      <td>TCGA-02-0026</td>\n",
       "      <td>[6.5582, 48.0798, 61.4221, 20.8733, 2.0219, 39...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9052</th>\n",
       "      <td>TCGA-02-0033</td>\n",
       "      <td>[2.301, 27.3171, 27.362, 24.4932, 1.2572, 9.90...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9053</th>\n",
       "      <td>TCGA-02-0038</td>\n",
       "      <td>[2.821, 19.328, 29.0524, 6.3993, 1.488, 91.915...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19553</th>\n",
       "      <td>TCGA-ZS-A9CD</td>\n",
       "      <td>[29.1061, 19.0332, 12.5332, 2.8365, 0.5151, 29...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19554</th>\n",
       "      <td>TCGA-ZS-A9CE</td>\n",
       "      <td>[8.6253, 26.0706, 11.7897, 1.9057, 0.9133, 48....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19556</th>\n",
       "      <td>TCGA-ZS-A9CG</td>\n",
       "      <td>[79.5957, 18.0213, 16.5968, 2.894, 0.8658, 6.6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19557</th>\n",
       "      <td>TCGA-ZT-A8OM</td>\n",
       "      <td>[0.5088, 15.2741, 11.3042, 2.7836, 1.0875, 2.9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19559</th>\n",
       "      <td>TCGA-ZX-AA5X</td>\n",
       "      <td>[3.8986, 59.159, 42.7654, 7.7885, 2.0867, 131....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9594 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       case_barcode                                     tpm_unstranded\n",
       "9049   TCGA-02-0003  [4.5317, 23.5094, 36.0164, 8.1348, 1.204, 38.3...\n",
       "9050   TCGA-02-0016  [24.0906, 44.8204, 23.6335, 12.3301, 3.2369, 6...\n",
       "9051   TCGA-02-0026  [6.5582, 48.0798, 61.4221, 20.8733, 2.0219, 39...\n",
       "9052   TCGA-02-0033  [2.301, 27.3171, 27.362, 24.4932, 1.2572, 9.90...\n",
       "9053   TCGA-02-0038  [2.821, 19.328, 29.0524, 6.3993, 1.488, 91.915...\n",
       "...             ...                                                ...\n",
       "19553  TCGA-ZS-A9CD  [29.1061, 19.0332, 12.5332, 2.8365, 0.5151, 29...\n",
       "19554  TCGA-ZS-A9CE  [8.6253, 26.0706, 11.7897, 1.9057, 0.9133, 48....\n",
       "19556  TCGA-ZS-A9CG  [79.5957, 18.0213, 16.5968, 2.894, 0.8658, 6.6...\n",
       "19557  TCGA-ZT-A8OM  [0.5088, 15.2741, 11.3042, 2.7836, 1.0875, 2.9...\n",
       "19559  TCGA-ZX-AA5X  [3.8986, 59.159, 42.7654, 7.7885, 2.0867, 131....\n",
       "\n",
       "[9594 rows x 2 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_grouped_expressions_df['case_barcode'].str.contains('TCGA')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "trusted": true
   },
   "outputs": [],
   "source": [
    "grouped_expressions_df['tpm_unstranded'].apply(len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.729299Z",
     "iopub.status.idle": "2025-10-15T13:44:53.729655Z",
     "shell.execute_reply": "2025-10-15T13:44:53.729474Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "genes_mutations = \"\"\"'ABL1', 'ADAM15', 'AGT', 'ANXA2', 'APC', 'APP', 'APPL1', 'ARAF',\n",
    "           'ARHGAP9', 'ATR', 'BCLAF1', 'BGN', 'BMX', 'BPTF', 'BTRC', 'CBL',\n",
    "           'CBLB', 'CCR2', 'CD244', 'CD300LF', 'CD9', 'CDC25B', 'CDH1',\n",
    "           'CEP70', 'CGN', 'CHN2', 'CLTC', 'CNTRL', 'COL18A1', 'COL1A2',\n",
    "           'COL6A3', 'COLEC12', 'CP', 'CPT2', 'CREBBP', 'CRK', 'CSF2RB',\n",
    "           'CSF3R', 'CTNNA1', 'CTTN', 'CXorf21', 'CYTH4', 'DENND3', 'DENND4C',\n",
    "           'DLG1', 'DNAJC6', 'DOCK1', 'DOCK8', 'DSG2', 'DUSP10', 'DUSP19',\n",
    "           'DUSP26', 'DUSP6', 'DVL2', 'DVL3', 'EGFR', 'EMILIN1', 'EN1',\n",
    "           'EP300', 'EPC1', 'EPM2A', 'EPN1', 'EPN2', 'EPS8', 'ERRFI1', 'EYA1',\n",
    "           'EYA2', 'F10', 'F8', 'FASN', 'FASTKD5', 'FBF1', 'FBN1', 'FBXO25',\n",
    "           'FER', 'FGF10', 'FGF5', 'FGFR2', 'FGFR4', 'FLT4', 'FOXK1', 'FOXO3',\n",
    "           'FOXO4', 'FRS2', 'FUT7', 'GAB2', 'GAB3', 'GATA2', 'GFI1', 'GLI2',\n",
    "           'GNA11', 'GNAQ', 'HAL', 'HIF1A', 'HNF4A', 'HRC', 'HSP90AA1',\n",
    "           'HSPG2', 'IKZF1', 'IL17RA', 'IL1B', 'INPP5D', 'INPPL1', 'INTU',\n",
    "           'ITGA1', 'ITGA11', 'ITGA2', 'ITGAV', 'ITGB2', 'ITGB3', 'JAK1',\n",
    "           'JAK3', 'KDR', 'KIT', 'KL', 'KLB', 'KLHL12', 'LAPTM4B', 'LGMN',\n",
    "           'LRFN1', 'LRP1', 'LRP6', 'LRRFIP1', 'LYN', 'MAPK11', 'MAPK12',\n",
    "           'MEF2C', 'MET', 'MFGE8', 'MMP2', 'MPDZ', 'MPO', 'MRC2', 'MS4A6A',\n",
    "           'MTM1', 'MTMR3', 'MTOR', 'MYB', 'MYO1C', 'MYO1F', 'MYO1G', 'NAA15',\n",
    "           'NCK1', 'NEBL', 'NEDD4', 'NFKB1', 'NIPSNAP3B', 'NOS3', 'NRP1',\n",
    "           'NRP2', 'NTRK1', 'ONECUT1', 'ORAI1', 'OTUD7B', 'PAG1', 'PALLD',\n",
    "           'PCDH19', 'PDCD6IP', 'PDGFRA', 'PEX5', 'PHLDB2', 'PIK3C2A',\n",
    "           'PIK3C2B', 'PIK3CA', 'PIK3CB', 'PIK3R3', 'PIP4K2A', 'PLCE1',\n",
    "           'PLD4', 'PLK1', 'PLXNA1', 'PPFIA3', 'PRKCA', 'PRKCB', 'PRKCG',\n",
    "           'PRKCH', 'PRLR', 'PTEN', 'PTK2', 'PTP4A2', 'PTPN11', 'PTPN13',\n",
    "           'PTPN14', 'PTPN4', 'PTPN6', 'PTPRB', 'PTPRC', 'PTPRD', 'PTPRF',\n",
    "           'PTPRJ', 'PTPRK', 'PTPRN2', 'PTPRQ', 'PTPRS', 'PTPRU', 'RASA1',\n",
    "           'RASA2', 'RCSD1', 'RECK', 'REL', 'RET', 'ROCK1', 'RYR1', 'S1PR2',\n",
    "           'S1PR5', 'SCFD1', 'SERPINB10', 'SGK1', 'SH2B1', 'SH2B2', 'SH2B3',\n",
    "           'SH2D2A', 'SH3KBP1', 'SHC1', 'SHC2', 'SHC3', 'SLC46A1', 'SMAD4',\n",
    "           'SNAI2', 'SOS1', 'SOS2', 'SPRED2', 'SREBF1', 'SSH2', 'STAT1',\n",
    "           'STAT2', 'STAT3', 'STAT4', 'TFAP2A', 'TFAP2B', 'TGFBR2', 'TGIF1',\n",
    "           'THOC5', 'TJP1', 'TNK1', 'TNS1', 'TP53', 'TPD52', 'TRAF3IP1',\n",
    "           'TRAF4', 'TRIP6', 'TXK', 'TYK2', 'UBA5', 'USP20', 'VAV2', 'VAV3',\n",
    "           'VEGFA', 'VTN', 'WASF2', 'WDR35', 'WDR74', 'YY1', 'ZBTB34', 'ZEB1',\n",
    "           'ZEB2', 'ZHX2', 'ZZEF1'\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.730560Z",
     "iopub.status.idle": "2025-10-15T13:44:53.731010Z",
     "shell.execute_reply": "2025-10-15T13:44:53.730803Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "filtered_df.to_csv('filtered_df.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.731919Z",
     "iopub.status.idle": "2025-10-15T13:44:53.732348Z",
     "shell.execute_reply": "2025-10-15T13:44:53.732190Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "data_mutations = []\n",
    "for dataset in filtered_df['Full ID_mut']:\n",
    "        query = f\"\"\"\n",
    "        SELECT\n",
    "            m.case_barcode,\n",
    "            m.Hugo_Symbol,\n",
    "            m.Start_position,\n",
    "            m.End_position,\n",
    "            m.Chromosome,\n",
    "            m.Variant_Type,\n",
    "            m.Reference_Allele,\n",
    "            m.Tumor_Seq_Allele2,\n",
    "            m.Tumor_Seq_Allele1,\n",
    "            m.Gene,\n",
    "            m.Exon,\n",
    "            m.Strand,\n",
    "            m.primary_site\n",
    "        FROM\n",
    "        {dataset} AS m\n",
    "        WHERE m.Hugo_Symbol in ({genes})\n",
    "        ORDER BY m.Hugo_Symbol         \n",
    "        ;\n",
    "        \"\"\"\n",
    " \n",
    "        try: \n",
    "            query_job = client.query(query)\n",
    "            result_df_mutations = query_job.to_dataframe()\n",
    "            result_df_mutations['dataset'] = dataset\n",
    "            data_mutations.append(result_df_mutations)\n",
    "        except Exception as e:\n",
    "            print(e)\n",
    "            print(dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.733065Z",
     "iopub.status.idle": "2025-10-15T13:44:53.733383Z",
     "shell.execute_reply": "2025-10-15T13:44:53.733256Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "#Index(['primary_site', 'file_gdc_id', 'project_short_name', 'Variant_Type',\n",
    "#       'Variant_Classification', 'Reference_Allele', 'Tumor_Seq_Allele1',\n",
    "#       'Tumor_Seq_Allele2', 'Gene', 'Feature', 'Hugo_Symbol', 'Exon', 'Start',\n",
    "#       'End', 'Genomic_Bin', 'Strand', 'dataset', 'Chromosome',\n",
    "#       'Mutational_Motif'],\n",
    "#      dtype='object')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.733982Z",
     "iopub.status.idle": "2025-10-15T13:44:53.734334Z",
     "shell.execute_reply": "2025-10-15T13:44:53.734190Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "\n",
    "df_mutations = pd.concat(data_mutations, ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.735166Z",
     "iopub.status.idle": "2025-10-15T13:44:53.735512Z",
     "shell.execute_reply": "2025-10-15T13:44:53.735371Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "df_mutations.to_parquet('mutations.parquet')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.737000Z",
     "iopub.status.idle": "2025-10-15T13:44:53.737493Z",
     "shell.execute_reply": "2025-10-15T13:44:53.737294Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "df_expressions.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.738226Z",
     "iopub.status.idle": "2025-10-15T13:44:53.738541Z",
     "shell.execute_reply": "2025-10-15T13:44:53.738393Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "grouped_expression_df = df_expressions.groupby('case_barcode').agg({\n",
    "    'gene_name': list,\n",
    "    'tpm_unstranded': list,\n",
    "}).reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.739250Z",
     "iopub.status.idle": "2025-10-15T13:44:53.739564Z",
     "shell.execute_reply": "2025-10-15T13:44:53.739412Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "grouped_expression_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.740294Z",
     "iopub.status.idle": "2025-10-15T13:44:53.740614Z",
     "shell.execute_reply": "2025-10-15T13:44:53.740454Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "grouped_expression_df['case_barcode'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.741326Z",
     "iopub.status.idle": "2025-10-15T13:44:53.741626Z",
     "shell.execute_reply": "2025-10-15T13:44:53.741485Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "grouped_mutations_df = df_mutations.groupby('case_barcode').agg({\n",
    "    'Hugo_Symbol': list,\n",
    "    'primary_site': 'first'\n",
    "    #'Variant_Type': 'count',\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.742321Z",
     "iopub.status.idle": "2025-10-15T13:44:53.742604Z",
     "shell.execute_reply": "2025-10-15T13:44:53.742491Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "grouped_mutations_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.743314Z",
     "iopub.status.idle": "2025-10-15T13:44:53.743672Z",
     "shell.execute_reply": "2025-10-15T13:44:53.743531Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "combined_df = pd.merge(grouped_mutations_df, grouped_expression_df, on=\"case_barcode\", suffixes=('_mutation', '_expression'))\n",
    "combined_df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution": {
     "iopub.status.busy": "2025-10-15T13:44:53.744371Z",
     "iopub.status.idle": "2025-10-15T13:44:53.744646Z",
     "shell.execute_reply": "2025-10-15T13:44:53.744534Z"
    },
    "trusted": true
   },
   "outputs": [],
   "source": [
    "combined_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "trusted": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kaggle": {
   "accelerator": "none",
   "dataSources": [
    {
     "datasetId": 7030230,
     "sourceId": 11458791,
     "sourceType": "datasetVersion"
    }
   ],
   "dockerImageVersionId": 30886,
   "isGpuEnabled": false,
   "isInternetEnabled": true,
   "language": "python",
   "sourceType": "notebook"
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
